addons/sms/tests/test_sms_template_access_rights.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.tests.common import SavepointCase, new_test_user
+from odoo.exceptions import AccessError
+
+
+class TestSmsTemplateAccessRights(SavepointCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.sms_manager = new_test_user(cls.env, login='sms_manager', groups='base.group_system')
+        cls.sms_user = new_test_user(cls.env, login='sms_user', groups='base.group_user')
+
+    def create_sms_template(self, user, model):
+        return self.env['sms.template'].with_user(user).create(dict({
+            'name': 'SMS Template ' + model.name,
+            'body': 'Body Test',
+            'model_id': model.id,
+        }))
+
+    def read_sms_template(self, user, templates, field):
+        data = templates.with_user(user).read([field])
+        if len(templates) == 1:
+            return data[0][field]
+        mapped_data = {record['id']: record for record in data}
+        return [mapped_data[template_id][field] for template_id in templates.ids]
+
+    def test_access_rights_sms_template(self):
```

Comment:
I would create 2 tests.
One for the base.group_user and one for the base.group_admin.

That's the way we usually do it and you will be able to use the `@users` decorator on your methods to make your life easier.

Reply:
done